src/main/java/org/openstreetmap/atlas/checks/validation/RoadNameGapCheck.java

```diff
@@ -0,0 +1,134 @@
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public RoadNameGapCheck(final Configuration configuration)
+    {
+        super(configuration);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Edge && Edge.isMasterEdgeIdentifier(object.getIdentifier())
+                && TagPredicates.IS_HIGHWAY_NOT_LINK_TYPE.test(object)
+                && TagPredicates.VALID_HIGHWAY_TAG.test(object)
+                && HighwayTag.isCarNavigableHighway(object) && !JunctionTag.isRoundabout(object);
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Edge edge = (Edge) object;
+        final Set<Edge> inEdges = edge.inEdges().stream().filter(this::validCheckForObject)
+                .collect(Collectors.toSet());
+        final Set<Edge> outEdges = edge.outEdges().stream().filter(this::validCheckForObject)
+                .collect(Collectors.toSet());
+
+        if (inEdges.isEmpty() || outEdges.isEmpty())
+        {
+            return Optional.empty();
+        }
+
+        final Set<String> edgeNames = findInEdgeOutEdgeMatchingName(inEdges, outEdges);
+
+        // Return empty if there is no pair of in edge and out edge with same name.
+        if (edgeNames.isEmpty())
+        {
+            return Optional.empty();
+        }
+
+        // Create flag when we have in edge and out edge with same name but intermediate edge
+        // doesn't have a name.
+        if (!edge.getName().isPresent())
+        {
+            final String instruction = "Edge name is empty.";
+            return Optional.of(createFlag(object, instruction));
+        }
+
+        // Create flag when in edge and out edge name tag matches and intermediate edge has
+        // different tag name.
+        final Optional<String> edgeName = edge.getName();
+        if (edgeName.isPresent() && !edgeNames.contains(edgeName.get()))
+        {
+            final String instruction = "Edge name is different from in edge name and out edge name.";
```

Comment:
Adding both the flagged Edge's name and the name of the mismatched in/out Edges will be useful for community mappers and debugging.

Reply:
`getLocalizedInstruction` is a useful function for handling instructions.

You can declare your instructions as a static global variables [like this](https://github.com/osmlab/atlas-checks/blob/dev/src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidTagsCheck.java#L36)
and fetch the values [like this](https://github.com/osmlab/atlas-checks/blob/dev/src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidTagsCheck.java#L145)